Cannot compile for targets without 64-bit atomics when using nightly feature

Some platforms are missing `AtomicU64` and `AtomicI64` (because they are 32-bit only). I want to use this crate in my embedded project targeting a cortex-m 32-bit microcontroller. When I enable the nightly feature, I get this compile error:

```
error[E0432]: unresolved imports `core::sync::atomic::AtomicI64`, `core::sync::atomic::AtomicU64`
  --> /home/jamadazi/.cargo/registry/src/github.com-1ecc6299db9ec823/atomic-0.4.5/src/ops.rs:17:27
   |
17 |     AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicU16, AtomicU32, AtomicU64, AtomicU8,
   |                           ^^^^^^^^^ no `AtomicI64` in `sync::atomic` ^^^^^^^^^ no `AtomicU64` in `sync::atomic`
   |
help: a similar name exists in the module
   |
17 |     AtomicI16, AtomicI32, AtomicI8, AtomicI8, AtomicU16, AtomicU32, AtomicU64, AtomicU8,
   |                           ^^^^^^^^
help: a similar name exists in the module
   |
17 |     AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicU16, AtomicU32, AtomicU8, AtomicU8,
   |                                                                      ^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `atomic`.
```

It seems like the `atomic` crate is trying to unconditionally use 64-bit atomics, which are not available on this target.

--- a/src/ops.rs
+++ b/src/ops.rs
@@ -12,10 +12,14 @@ use core::ops;
 use core::sync::atomic::Ordering;
 use fallback;
 
-#[cfg(feature = "nightly")]
-use core::sync::atomic::{
-    AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicU16, AtomicU32, AtomicU64, AtomicU8,
-};
+#[cfg(all(feature = "nightly", target_has_atomic = "8"))]
+use core::sync::atomic::{ AtomicI8, AtomicU8 };
+#[cfg(all(feature = "nightly", target_has_atomic = "16"))]
+use core::sync::atomic::{ AtomicI16, AtomicU16 };
+#[cfg(all(feature = "nightly", target_has_atomic = "32"))]
+use core::sync::atomic::{ AtomicI32, AtomicU32 };
+#[cfg(all(feature = "nightly", target_has_atomic = "64"))]
+use core::sync::atomic::{ AtomicI64, AtomicU64 };
 
 #[cfg(not(feature = "nightly"))]
 use core::sync::atomic::AtomicUsize;

